docs/concepts/bootstrap-vs-build.rst

```diff
@@ -0,0 +1,109 @@
+Bootstrap vs Build
+==================
+
+Fromager has two distinct modes of operation: **bootstrap** and **build**.
+Understanding the difference is key to using fromager effectively.
+
+Quick Comparison
+----------------
+
+.. list-table::
+   :header-rows: 1
+   :widths: 20 40 40
+
+   * - Aspect
+     - Bootstrap
+     - Build
+   * - **Scope**
+     - Entire dependency tree
+     - Single package
+   * - **Purpose**
+     - Discover and resolve all dependencies
+     - Compile source into wheel
+   * - **Recursion**
+     - Yes (processes dependencies)
+     - No (one package only)
+   * - **Input**
+     - Requirements file or package specs
+     - Package name + version + source URL
+   * - **Output**
+     - Dependency graph, build order, all wheels
+     - One wheel file
+
+Bootstrap Mode
+--------------
+
+The ``bootstrap`` command recursively discovers and builds all dependencies:
+
+.. code-block:: text
+
+   fromager bootstrap numpy
```

Comment:
I don't know how pedantic we want to be about making this set of steps accurate for a concept guide. If we want 100% accuracy, then we need a step after "download source" to extract the build dependencies and then steps to bootstrap all of those recursively.